### PR TITLE
Update title and name for all schemas. Fixes #501.

### LINKED ISSUE #501: Make title attribute user-friendly and Sentence case
**For which schema is a change/update being suggested?**

I would like to request an update to all metadata schemas.

**What should the change/update be?**

I would like to change all schema `title` attributes from lowercase, snake_case terms to user-friendly Sentence case terms.

_NEW_ I would like to add to all schema a `name` attribute to be the programmatic name of the schema, which should always be the name of the schema file.

**What new field(s) need to be changed/added?**

No new fields.

**Why is the change requested?**

Currently the `title` attribute is not used anywhere. This attribute would be a good place to store a user-friendly schema name to be used as spreadsheet tab names, to append to custom field names (_e.g._ to generate "Organoid ID" from the `biomaterial_id` field in the `organoid` schema), and for displaying in things like the ingest UI and Data Portal. This change would be a patch change to all schemas.

_NEW_ Eventually ingest will need a programmatic way to access the name of the schema (currently parsing the URI which is **not** correct), so including this programmatic name in a `name` attribute would satisfy this requirement.

### PR DESCRIPTION
**Pull request by malloryfreeberg**
- Added `name` attribute to all schemas with the programmatic name of the schema (is the file name)
- Updated `title` attribute in all schemas with user-facing name of the schema
- Updated a few schema `description` attributes for standardization
- Updated update_log.csv

### CODE REVIEW

File: docs/jsonBrowser/core.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ Property name | Description | Type | Required? | Example
  describedBy | The URL reference to the schema. | string | no |  |  |  | 
 schema_version | The version number of the schema in major.minor.patch format. | string | no | 4.6.1
 
-## biomaterial_core<a name='biomaterial_core'></a>
+## Biomaterial core<a name='Biomaterial core'></a>
```
[the remark this answers is not in the document]
Comment (daniwelter) [reply docs/jsonBrowser/core.md]: The markdown docs are automatically built at every commit - nothing we can do about that unless we dismantle the entire githook

[the remark this answers is not in the document]
Comment (hannes-ucsc) [reply docs/jsonBrowser/core.md]: What triggered the change in casing from lower case to upper case?

[the remark this answers is not in the document]
Comment (hannes-ucsc) [reply docs/jsonBrowser/core.md]: I guess the mechanics of the hook itself must have hook changed. I'm not sure a hook is the right mechanism for generating documentation. Typically, documentation is created as part of a CI build process. Doing it as a a hook may be convenient but has the drawback of polluting commits with noise, noise that can even cause conflicts as is currently the case in this PR.

**Comment by diekhans** [comment]
IMHO, this comes down to if the goal is to have the documentation committed or
generated after the fact.

After the fact generation is the traditional approach, were
users download builds.  It has advantages of ensuring
consistency and being able access a releases without requiring
all of the build tools to be installed.

In the github world, people tend to checkout to use and view the
markdown directly in github, requiring it to be checked in and
consistent with the code.  This does create more work when their
are conflicts.

Offering no opinion on which option is preferred, other than if
a change is desired, create a ticket.

File: docs/jsonBrowser/core.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (daniwelter) [reply docs/jsonBrowser/core.md]: The change in casing is due to the docs build script using the JSON title field in this place. As title changed from programmatic title to user-friendly title in this branch, the markdown doc changed accordingly.

**Comment by daniwelter** [comment]
@diekhans Thank you for your comments.

There was a lot of debate on when/how to build the markdown docs and the current pre-commit Githook isn't necessarily the optimal solution. Just for context, the following elements influenced the decision:
- The metadata repo has a wide range of contributors with different levels of technical ability, so the markdown generation script needed to be easy to run
- Lots of commits means the script needs to run regularly in order for the markdown to stay consistent with the json schemas
- Markdown docs need to be checked into the repo
- Even minor changes in markdown docs can cause merge conflicts

Checking in docs built by a CI process is a non-trivial problem and as we had limited time to generate a solution, a pre-commit Githook was the best compromise to quickly and regularly build the docs without committers having to actively remember to run a script every time they commit a change. The hook also helps with markdown merge conflicts as the committer can be confident that as long as their branch is up-to-date, their docs will also be the "canonical" version, and they can therefore resolve any conflicts directly in their favour. Any incorrectly resolved conflicts will also be rectified at the next commit.

As I said, the precommit Githook isn't necessarily the best solution here but it was the best compromise at the time. I would be happy to revisit with better options if anyone has any suggestions. I am however closing the discussion here as it is OT for the PR - we can transfer comments to a ticket if anyone would like to discuss doc generation optimisation.

**Comment by diekhans** [comment]
Thanks @daniwelter

Ticket created here: https://github.com/HumanCellAtlas/metadata-schema/issues/523

**Pull request merged**